### PR TITLE
README.md: Updated OpenBSD build instructions

### PR DESCRIPTION
**Pull request by ston1th**
* Removed the db package, as it is not required
* Added the libiconv package
* Updated the boost build instructions (added locale)
* Updated cppzmq to version 4.2.3 which does not require the symlink hack anymore

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -415,22 +417,19 @@ Build the cppzmq bindings.
...
 
 # Fetch cppzmq source
-ftp -o cppzmq-4.2.2.tar.gz https://github.com/zeromq/cppzmq/archive/v4.2.2.tar.gz
+ftp -o cppzmq-4.2.3.tar.gz https://github.com/zeromq/cppzmq/archive/v4.2.3.tar.gz
 
-# MUST output: (SHA256) cppzmq-4.2.2.tar.gz: OK
-echo "3ef50070ac5877c06c6bb25091028465020e181bbfd08f110294ed6bc419737d cppzmq-4.2.2.tar.gz" | sha256 -c
-tar xfz cppzmq-4.2.2.tar.gz
+# MUST output: (SHA256) cppzmq-4.2.3.tar.gz: OK
+echo "3e6b57bf49115f4ae893b1ff7848ead7267013087dc7be1ab27636a97144d373 cppzmq-4.2.3.tar.gz" | sha256 -c
```
Comment (moneromooo-monero) [inline README.md]: Shouldn't these have somehting like || exit 1 (can be in another PR as it's existing) ?

File: README.md [hunk 2]
```diff
@@ -396,16 +396,18 @@ ftp -o boost_1_64_0.tar.bz2 https://netcologne.dl.sourceforge.net/project/boost/
...
 
-# Fetch a boost patch, required for OpenBSD
-ftp -o boost.patch https://raw.githubusercontent.com/openbsd/ports/bee9e6df517077a7269ff0dfd57995f5c6a10379/devel/boost/patches/patch-boost_test_impl_execution_monitor_ipp
+# Fetch and apply boost patches, required for OpenBSD
+ftp -o boost_test_impl_execution_monitor_ipp.patch https://raw.githubusercontent.com/openbsd/ports/bee9e6df517077a7269ff0dfd57995f5c6a10379/devel/boost/patches/patch-boost_test_impl_execution_monitor_ipp
+ftp -o boost_config_platform_bsd_hpp.patch https://raw.githubusercontent.com/openbsd/ports/90658284fb786f5a60dd9d6e8d14500c167bdaa0/devel/boost/patches/patch-boost_config_platform_bsd_hpp
 cd boost_1_64_0
-patch -p0 < ../boost.patch
+patch -p0 < ../boost_test_impl_execution_monitor_ipp.patch
+patch -p0 < ../boost_config_platform_bsd_hpp.patch
```
Comment (moneromooo-monero) [inline README.md]: Since this is patching from external sources, this could do with checking a hash too (can be in another PR as this is existing).

File: README.md (hunk 1 above)
Comment (ston1th) [reply README.md]: Usually, yes, if this would be a shell script.

But since this is a step by step copy/paste guide, adding a `|| exit 1` would just log out the users session.

File: README.md (hunk 2 above)
Comment (ston1th) [reply README.md]: I'll add hashes to those.

File: README.md (hunk 1 above)
Comment (moneromooo-monero) [reply README.md]: Ah, fair point :)

**Comment by ston1th** [comment]
@moneromooo-monero I added the hashes. I hope you are fine with them being in this PR.

**Comment by moneromooo-monero** [comment]
Sure, that's fine, thanks.
